src/process_utility.c

```diff
@@ -728,8 +757,16 @@ process_vacuum(ProcessUtilityArgs *args)
 					if (hypertable_is_distributed(ht))
 						continue;
 
-					ctx.ht_vacuum_rel = vacuum_rel;
-					foreach_chunk(ht, add_chunk_to_vacuum, &ctx);
+					if (ht->fd.compressed)
```

Comment:
This code won't be run in the case of a database-wide analyze.  That case is handled by ts_get_all_vacuum_rels up above in the `stmt->rels == NIL` case.  You should modify that call to also add the the compressed chunk pairs (keep in mind that that call will iterate over both the compressed and uncompressed chunk tables).